testing/thor/index.js

```diff
@@ -0,0 +1,155 @@
+import * as mjolnir from "mjolnir";
+import * as axios from "axios";
+
+let private_key = null;
+let public_key = null;
+let user_address = null;
+let user_address_hex = "";
+
+let utxos = [];
+let inputs = [];
+let outputs = [];
+
+window.document.getElementById("store_pk").onclick = function(self) {
+    console.log("loading private key..")
+    let hex = document.getElementById("private_key_hex").value;
+    private_key = mjolnir.PrivateKey.from_hex(hex);
+    console.log("extracting public key..")
+    let public_key = private_key.public();
+    document.getElementById("public_key_hex").value = public_key.to_hex();
+    console.log("extracting address key..")
+    user_address = public_key.address();
+    user_address_hex = user_address.to_betch32();
+    document.getElementById("user_address").value = user_address.to_betch32();
+};
+
+window.document.getElementById("tx_add").onclick = function(self) {
+    var out_addr = document.getElementById("tx_out_address").value;
+    var out_value = document.getElementById("tx_out_value").value;
+    outputs.push({"address":out_addr, "value": out_value});
+    show_outputs();
+}
+
+window.document.getElementById("sign_btn").onclick = function(self) {
+   console.log("Creating a builder.");
+   var txbuilder = mjolnir.TransactionBuilder.new();
+   console.log("Adding inputs:");
+   inputs.forEach(function (input, i) {
+      console.log("  input: ", input);
+      var tx = mjolnir.TransactionId.from_hex(input.in_txid);
+      var utxo = mjolnir.UtxoPointer.new(tx, input.in_idx, BigInt(input.out_value));
```

Comment:
This is not idiomatic JS, can we bind to the JS class constructor?

Reply:
not sure if I can do that easily, the value is not a normal JS clas.